README.md

```diff
@@ -3,30 +3,70 @@ A GPU implementation of 2, 3 dimension type 1, 2 non-uniform FFT based on [FINUF
 
 This is a work from Melody Shih's internship at Flatiron Institute, advised by CCM project leader Alex Barnett.
 
+## Installation
 
-### Installation
- - Get this code and dependency -
+Note for most Python users, you may skip to the [Python Package](#Python-Package) section first,
+and consider installing from source if that solution is not adequate for your needs.
+
+ - Get this code -
  ```git clone https://github.com/flatironinstitute/cufinufft.git```
  - Review the `makefile`. - If you need to customize build settings, create and edit a `make.inc`.  Example:
    - To override the standard CUDA `/usr/local/cuda` location your `make.inc` should contain: ```CUDA_ROOT=/your/path/to/cuda```.
    - Two examples are provided, one for IBM machines (make.inc.power9), and another for the Courant Institute cluster (make.inc.CIMS).
  - Compile - ```make all -j```
- - Run a test code - ``` ./cufinufft2d1_test 2 128 128 10 1e-6```
+ - Run test codes - ```make check```
+
+### Library Installation
+
+It is up to the user to decide how exactly to link or otherwise install the libraries produced in `lib`.
+If you plan to use the Python wrapper you will minimally need to extend your `LD_LIBRARY_PATH`,
+such as with `export LD_LIBRARY_PATH=${PWD}/lib:${LD_LIBRARY_PATH}` or a more permanant installation
+path of your choosing.
+
+If you would like to always have this installation in your library path, you can add to your shell rc
+with something like the following:
+
+`echo "\n# cufinufft librarypath \nexport LD_LIBRARY_PATH=${PWD}/lib:${LD_LIBRARY_PATH}" >> ~/.bashrc`
+
+Because CUDA itself has similar library/path requirements, it is expected the user is somewhat familiar.
+If not, please ask, we might be able to help.
+
+### Python Wrapper
+
+For those installing from source, this code comes with a Python wrapper module `cufinufftpy`.
+Once you have successfully installed and tested the CUDA library
+you may run `make python` to manually install the additional Python package.
+
+### Python Package
+
+General Python users, or Python software packages which would like to automatically
+depend on cufinufftpy using `setuptools` may use a precompiled binary distribution.
+This totally avoids installing from source and managing libraries for supported systems.
+
+Because binary distributions are specific to both hardware and software,
+we currently only support systems covered by `manylinux2010` that are using
+CUDA 10.1, 10.2, or 11.0-rc with a compatible GPU. This is currently the most
+common arrangement.  If you have such a system, you may run:
+
+`pip install cufinufftpy`
```

Comment:
Is there going to be an option here to specify the CUDA version (like `pip install cufinufftpy[cuda102]` or something like that)?

Reply:
Not right now.  For now, I think it is sufficient to say something like `spam==1.0` is prebuilt for CUDA X `spam==1.23` is prebuilt for CUDA Y. Currently I expect the wheel to work for the release of CUDA that has been dominant for the last year (10), and the one that will be out for the next year or so (11).

If this becomes really popular and there is buy in to support it, I can decouple and make a distribution packaging matrix. This would be like `cufinufftpy==2.0+cuda120` or similar.  That is more work than I can justify right now.

Reply:
Ok that makes sense. The risk is that users running on clusters (and other machines for which they don't have root access) may not have a choice in their CUDA version. That being said, they can just compile their own for now.